src/dml/dml_helpers.cpp

```diff
@@ -52,6 +52,31 @@ static std::vector<ComPtr<IDXGIAdapter1>> EnumerateAdapters() {
         adapter_infos.emplace_back(std::move(adapter));
       }
     }
+  } else if (device_luid) {
+    // Find the adapter with the required LUID.
+    ComPtr<IDXGIAdapter1> adapter;
+    for (uint32_t adapter_index = 0; dxgi_factory->EnumAdapters1(adapter_index, &adapter) != DXGI_ERROR_NOT_FOUND; adapter_index++) {
+      // We can't assume the ordering of hardware and software adapters, so keep looping. This path should only execute on Windows 10
+      // version 1709 or earlier; IDD (e.g. remote desktop) adapters do not exist when taking this code path.
+      if (IsSoftwareAdapter(adapter.Get())) {
+        continue;
+      }
+
+      // Make sure that we are able to create the device
+      ComPtr<ID3D12Device> d3d12_device;
+      THROW_IF_FAILED(D3D12CreateDevice(adapter.Get(), D3D_FEATURE_LEVEL_11_0, IID_PPV_ARGS(&d3d12_device)));
```

Comment:
This enumeration code is the same as the code below. I think It's better to have a single `else` block, and to only have an `if (device_luid)` when it's time to actually push it to the `adapter_infos` vector.

Reply:
Thanks, I  updated it